src/Runner.Worker/JobExtension.cs

```diff
@@ -404,6 +405,18 @@ public async Task<List<IStep>> InitializeJob(IExecutionContext jobContext, Pipel
                                                                           data: (object)jobHookData));
                     }
 
+                    // Register custom image creation post-job step if the "snapshot" token is present in the message.
+                    var snapshotRequest = templateEvaluator.EvaluateJobSnapshotRequest(message.Snapshot, jobContext.ExpressionValues, jobContext.ExpressionFunctions);
```

Comment:
i think this needs to move up before we calling `// Register Job Completed hook if the variable is set`

basically, snapshot first, then `job_complete_hook`

Reply:
Thanks, I think you're right. I'll go ahead and move that up.